helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1258,12 +1258,12 @@ protected void checkLiveInstancesObservation(List<LiveInstance> liveInstances,
           if (lastInstances == null || !lastInstances.containsKey(instance)) {
             try {
               manager.addCustomizedStateRootChangeListener(this, instance);
-              logger.info(manager.getInstanceName() + " added customized root change listener for"
-                  + " " + instance
-                  + ", listener: " + this);
+              logger.info(manager.getInstanceName() + " added root path listener for customized "
```

Comment:
I understand you may want to keep logging as original. But since you are changing the logging code here, I suggest changing it to use parameterized logging: `logger.info("Hello {}", world);`. This parameterized logging has benefits of reduce string concatenation overhead when logging level info is not enabled, and easier to read in the code. Same as the other places in this PR.
Nit, I realized this for loop needs unindent.

Reply:
`+ "state change for" + " "`
Can we please refine the format as well.